### DISCUSSION TITLE
cannot render the text in SSR with nextjs?

### DISCUSSION THREAD

**Discussion by blackrabbit944** [facebook/lexical#2880] [General]
I am working on a  post website which i used the lexical for the rich text editor.
but i found when i trying to render the SSR website , the lexical cannot render the HTML before the javascript runing.

`    const initialConfig = {
        namespace: 'editor', 
        theme,
        onError,
        readOnly,
        nodes : [...nodes]
    };
`
`
    const editorStateRef = useRef();
`
`
    if (initialEditorState) {
        initialConfig['editorState'] = initialEditorState;
    }
` 

here is some code of my website, but seems it need render the html with javascript runing in client. is there any way to use SSR with lexical ? thanks

**Comment by acywatson** [comment]
Check out @lexical/headless

**Comment by PedroJBrito** [comment]
Hi,

I'm facing the same challenge. I've got a html string and I'm struggling on how can I map this string to the initialState that's injected on LexicalComposer. I suppose I need to make the parser but DOMParser is only available on client side. I've tried JSDOM but then I have some conflicts importing this library in a component that's used on client side. Have anyone succeed loading text in SSR? 

Thank you for your time.

**Reply by PedroJBrito** [reply]
@blackrabbit944 Can you help me with this? Did you manage how to solve this problem?
